src/app/clusters/scenes/SceneTableImpl.h

```diff
@@ -0,0 +1,244 @@
+///        The implementation of SerializeSave and ApplyScene were omitted and must be implemented in a way that
+///        is compatible with the SerializeAdd output in order to function with the Default Scene Handler.
+///        It is worth noting that this implementation is very memory consuming. In the current worst case,
+///        (Color control cluster), the Extension Field Set's value pair list TLV occupies 99 bytes of memory
+class DefaultSceneHandlerImpl : public scenes::SceneHandler
+{
+public:
+    static constexpr uint8_t kMaxValueSize = 4;
+    static constexpr uint8_t kMaxAvPair    = 15;
+
+    DefaultSceneHandlerImpl() = default;
+    ~DefaultSceneHandlerImpl() override{};
+
+    /// @brief From command AddScene, allows handler to filter through clusters in command to serialize only the supported ones.
+    /// @param endpoint[in] Endpoint ID
+    /// @param extensionFieldSet[in] ExtensionFieldSets provided by the AddScene Command, pre initialized
+    /// @param cluster[out]  Cluster in the Extension field set, filled by the function
+    /// @param serialisedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR SerializeAdd(EndpointId endpoint,
+                                    const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
+                                    ClusterId & cluster, MutableByteSpan & serialisedBytes) override
+    {
+        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList;
+        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair;
+        TLV::TLVWriter writer;
+        TLV::TLVType outer;
+
+        uint8_t pairCount  = 0;
+        uint8_t valueBytes = 0;
+
+        VerifyOrReturnError(SupportsCluster(endpoint, extensionFieldSet.clusterID), CHIP_ERROR_INVALID_ARGUMENT);
+
+        cluster = extensionFieldSet.clusterID;
+
+        auto pair_iterator = extensionFieldSet.attributeValueList.begin();
+        while (pair_iterator.Next() && pairCount < kMaxAvPair)
+        {
+            aVPair                          = pair_iterator.GetValue();
+            mAVPairs[pairCount].attributeID = aVPair.attributeID;
+            auto value_iterator             = aVPair.attributeValue.begin();
+
+            valueBytes = 0;
+            while (value_iterator.Next() && valueBytes < kMaxValueSize)
+            {
+                mValueBuffer[pairCount][valueBytes] = value_iterator.GetValue();
+                valueBytes++;
+            }
+            // Check we could go through all bytes of the value
+            VerifyOrReturnError(value_iterator.Next() == false, CHIP_ERROR_BUFFER_TOO_SMALL);
+            mAVPairs[pairCount].attributeValue = mValueBuffer[pairCount];
+            mAVPairs[pairCount].attributeValue.reduce_size(valueBytes);
+            pairCount++;
+        }
+
```

Comment:
Where is the `pair_iterator.GetStatus()` check?  There has to be a GetStatus() check!